### PR TITLE
Updating 'SonicWall Firewall' Solution to version 3.1.0

### PR DESCRIPTION
**Pull request by jaimeesc**
Change(s):
   - Updating 'SonicWall Firewall' Solution to version 3.1.0
   - Adding Analytic Rules, a Hunting Query, and a Workbook.

   Reason for Change(s):
   - Submitting parsers and other content to the repository.

   Version Updated:
   - Yes. Updating to version 3.1.0. Added 2 Analytic Rules, 1 Hunting Query, and 1 Workbook.

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @jaimeesc,
1. Please resolve validation error for AllowedInboundSSHTelnetRDPConnections.

2. Please add a custom table at '.script/tests/KqlvalidationsTests/CustomTables' and 
    `.script/tests/KqlvalidationsTests/CustomFunctions` with name as **ASimNetworkSessionSonicWallFirewall**. You can refer to 
     any of the tables from the folder for more clarification.

**Comment by jaimeesc** [comment]
> 2\. script/tests/KqlvalidationsTests/CustomTables

Thanks for your response! Just one question I haven't found an answer to. Should I get the schema from the parser and include all columns (even default ones), or custom columns? The examples seem to have examples of both.

**Comment by jaimeesc** [comment]
The validations failed and seems unrelated to the changes since the last validation run. Going to try closing/re-opening the PR to kick off the validations again.

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, I am seeing this kind of Validation error for the first time. Please lend me some time to examine it.

**Comment by v-prasadboke** [comment]
> > 2. script/tests/KqlvalidationsTests/CustomTables
> 
> Thanks for your response! Just one question I haven't found an answer to. Should I get the schema from the parser and include all columns (even default ones), or custom columns? The examples seem to have examples of both.

Yes Columns used in parsers, should be added to the table. and the validation errors are visible now. 
please have a look.

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, All checks are green. I'll review this PR and get back to you by 26 December, 2023.

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, can you share sample data to test the content of the solution.

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, please share the sample data and add workbook metadata to this file https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/WorkbooksMetadata.json

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, can you please share the sample data and add the workbook metadata to the same.

**Comment by jaimeesc** [comment]
> Hello @jaimeesc, can you share sample data to test the content of the solution.

My apologies for the delay. The sample data is in the PR for the ASIM parser.
https://github.com/Azure/Azure-Sentinel/pull/9592

I will add the sample data to this PR as well.

**Comment by jaimeesc** [comment]
> Hello @jaimeesc, please share the sample data and add workbook metadata to this file https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/WorkbooksMetadata.json

I assume I need to duplicate and include the workbook JSON file as well? I did not see any other entries in the metadata file that pointed to the Solutions folder.

**Comment by jaimeesc** [comment]
I see all validations passed. Should be good to go. What happens next?

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, Thanks for commiting the required changes. I'll take a look at it and come back to you by 08 January, 2024.

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, There is no such any parser at mentioned location in Hunting Query Description 
image

**Comment by jaimeesc** [comment]
> Hello @jaimeesc, There is no such any parser at mentioned location in Hunting Query Description image

Hi,

I was asked to split the parsers into different PRs, so this PR does not contain the parsers.
https://github.com/Azure/Azure-Sentinel/pull/9592
https://github.com/Azure/Azure-Sentinel/pull/9593

**Comment by v-prasadboke** [comment]
I guess, We would have to get those above mentioned PR merged first. Solution gets deployed but as these parser is not tested yet by the respective reviewer from the above PR's we cannot proceed for this one.

**Comment by jaimeesc** [comment]
Any chance you can help look at the other 2 PRs? I haven't received a response and they've been validated for much longer than this one :).

**Comment by v-prasadboke** [comment]
Sure @jaimeesc, I'll ask the respective reviewer.

**Comment by jaimeesc** [comment]
Just a quick FYI. I am making the recommended changes on the other PRs. I'll be updating this one as well to reflect some differences in the parsers.

**Comment by v-prasadboke** [comment]
Sure @jaimeesc, No worries

**Comment by jaimeesc** [comment]
I'm working on some updates to the queries since I've made some large changes to the parsers in my other 2 PRs. I'll be updating this PR soon.

**Comment by jaimeesc** [comment]
More changes coming... I'm updating on the queries in the workbook in accordance with the parser updates. I will also look at the WorkbooksMetadata.json conflict.

**Comment by v-prasadboke** [comment]
Thanks for the update @jaimeesc

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, 
Its been too long we havent heard anything from you yet. Can you please share any updates regarding commits

**Comment by jaimeesc** [comment]
> Hello @jaimeesc, Its been too long we havent heard anything from you yet. Can you please share any updates regarding commits

Hello. Per my conversation with Atul Yadav, we're focusing on one PR at a time as each PR will have many of the same points to address. I updated https://github.com/Azure/Azure-Sentinel/pull/9592 last week. Hopefully we're close to accepting that one so I can move on to the next.

**Comment by v-prasadboke** [comment]
If its ok can you move this PR to draft state

**Comment by v-prasadboke** [comment]
Thank you @jaimeesc for moving this PR to draft state

**Comment by jaimeesc** [comment]
The Network Session and Web Session PRs have been updated. I'll update this PR soon.

**Comment by jaimeesc** [comment]
SonicWall's Network Session parser has merged. I'm waiting for review on the Web Session parser PR.
The Analytic Rules and Hunting Queries in this PR use either CommonSecurityLog or the ASimNetworkSessionSonicWallFirewall parser.
FYI, the Workbook has queries that use the Web Session parser, so we may see an error validating that.

**Comment by jaimeesc** [comment]
I replaced the contents of my WorkbooksMetadata.json file with the current master's contents. I then added my changes to the file. There should not be a conflict.

image

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, 
Thanks for marking this PR as ready for review. We will investigate the PR and get back to you by 14 March, 2024

**Comment by v-prasadboke** [comment]
Hello @jaimeesc. Sorry for the inconvenience. But we are facing react error in the content hub of azure portal. 
We are working on it and get back to you.

**Comment by v-prasadboke** [comment]
Hello @jaimeesc, We are facing an issue while custom deploying the maintemplate. 
Content hub is throwing an react error. 

Can you try to deploy the maintemplate from your end and check if its working. 
Thanks.

**Comment by jaimeesc** [comment]
I was able to deploy it without the createUiDefition file (custom deployment, pasting in the contents of mainTemplate. I'm looking into the problem with the location when using createUiDefinition.

**Comment by v-prasadboke** [comment]
Try to access the solution after deploying. 
Go to content hub click on your solution name. 

You will find manage button in the lower side of right pane. 
Check if you can see the content of the solution. 
Also please share a screenshot of the same.

**Comment by jaimeesc** [comment]
It deployed successfully (without the createUiDefition.json file). I can see the solution in content hub.
I'm still trying to work through the issue deploying with createUiDefinition.
image

When I clicked manage, I got this error. Is this caused by the template?
image

I can see the workbook template is there.
image

The hunting query is there.
image

The analytic rules are there.
image

The data connectors are there.
image

**Comment by jaimeesc** [comment]
I noticed the solution's version number displayed right before the react error is 2.0.6. The current version posted to GitHub is 3.0.0. I double-checked by deploying another Sentinel workspace and checking the content hub. I installed 2.0.6 from the content hub and clicked manage. No react error, but only one connector is installed (as expected for version 2.0.6). The react error seems to occur when I deploy a newer template than 2.0.6, but check content hub and it tries to load version 2.0.6 information. Could the issue be that the content references the solution (which doesn't exist in that version) or the solution references a data connector that was renamed post-2.0.6?

**Comment by jaimeesc** [comment]
Version 3.0.0 will be published to the Marketplace soon to replace 2.0.6. I think that will address the react error.
I believe I've addressed the issue between the maintemplate an createuidefinition. I can deploy successfully with and without the custom UI.

**Comment by v-prasadboke** [comment]
This issue I'm facing is for v 3.0.0

If you custom deploy the maintemplate via "deploy a custom template" on azure portal and head towards LA with deployed maintemplate 
you see this error after clicking on the manage button

**Comment by jaimeesc** [comment]
3.0.0 (the current version on github) should publish to the content hub soon. If you notice, the version on the content hub is older. The difference between 2.0.6/2.0.7 and 3.0.0 is the change to the data connector for the upcoming Log Analytics agent retirement. Based on what I've seen, it fails to load the data connector info that is expected by version 2.0.6/2.0.7. I suspect that when 3.0.0 publishes, 3.1.0 will not experience this error. Let's please see if that resolves it in the next day or so when it gets published.

**Comment by jaimeesc** [comment]
3.0.0 is now published in the content hub. Please go ahead and try again. I was able to deploy 3.1.0 from the maintemplate and can manage from the content hub. 

image

File: Solutions/SonicWall Firewall/Hunting Queries/OutboundSSHConnections.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+id: dedb8fb9-3caa-4b00-ae88-1898eed78917
+name: Outbound SSH/SCP Connections
+description: |
+  'This query looks for outbound SSH/SCP connections identified by the expected port number (22) or by the SonicWall Deep Packet Inspection services.
```
Comment (v-prasadboke) [inline Solutions/SonicWall Firewall/Hunting Queries/OutboundSSHConnections.yaml:4]: Hi @jaimeesc, 
Everything is good. I have tested the solution. Just we need this Hunting Query Description below 255 characters.

**Comment by v-prasadboke** [comment]
Proceed with the above mentioned changes and we are good to merge this PR

**Review by v-prasadboke — APPROVED** [review]

**Pull request merged**